### PR TITLE
Update CI to OSX 10.12

### PR DESCRIPTION
**Pull request by StephenBonikowsky**

### CODE REVIEW

**Comment by StephenBonikowsky** [comment]
@dotnet-bot test ci please

File: netci.groovy [hunk 1]
```diff
@@ -22,7 +22,7 @@ def osGroupMap = ['Ubuntu':'Linux',
                   'RHEL7.2': 'Linux']
 
 // Map of osName -> nuget runtime
-def targetNugetRuntimeMap = ['OSX' : 'osx.10.10-x64',
+def targetNugetRuntimeMap = ['OSX' : 'osx.10.12-x64',
```
Comment (weshaggard) [inline netci.groovy]: I think you need to update OSX to OSX.10.12 throughout this file. cc @mmitche

Comment (mmitche) [inline netci.groovy]: Yes, the setMachineAffinity call being the critical one here (otherwise it will run on 10.11)

Comment (StephenBonikowsky) [inline netci.groovy]: Updated.

**Comment by StephenBonikowsky** [comment]
@dotnet-bot test ci please

File: netci.groovy [hunk 2]
```diff
@@ -36,7 +36,7 @@ def osShortName = ['Windows 10': 'win10',
                    'Windows 7' : 'win7',
                    'Windows_NT' : 'windows_nt',
                    'Ubuntu14.04' : 'ubuntu14.04',
-                   'OSX' : 'osx',
```
Comment (weshaggard) [inline netci.groovy]: It doesn't look like this shortname map is used anywhere.

Comment (StephenBonikowsky) [inline netci.groovy]: Looks like none of them are used anywhere that I can see.

**Pull request merged**